test/integration/integration_test.go

```diff
@@ -0,0 +1,147 @@
+					app.WithVersion("test"), app.WithOutputDirectory(outDirRelPath),
+					app.WithLogger(log.New(GinkgoWriter, "", log.LstdFlags)),
+				)
+
+				err := a.Run()
+				Expect(err).ToNot(HaveOccurred())
+
+				fileName := interfaceName + ".go"
+				genFilePath := filepath.Join(generatedDirRelPath, snakeCase(fileName))
+				goldFilePath := filepath.Join(goldFilesDirRelPath, snakeCase(fileName))
+
+				compareFileContent(genFilePath, goldFilePath)
+			},
+			Entry(nil, "SingleMethodNoArgsNoReturns"),
+			Entry(nil, "SingleMethodWithArgNoReturns"),
+			Entry(nil, "SingleMethodWithArgAndReturn"),
+			Entry(nil, "ContextFirstArgument"),
+			Entry(nil, "ReturnsSingleError"),
+			Entry(nil, "ReturnsMultipleValuesWithError"),
+			Entry(nil, "ReturnsMultipleValuesNoError"),
+			Entry(nil, "ContextFirstArgumentReturnsError"),
+			Entry(nil, "ContextFirstArgumentReturnsErrorNamedVars"),
+			Entry(nil, "TwoMethodsOneWithContextAndError"),
+			Entry(nil, "MapInArgAndReturn"),
+			Entry(nil, "SliceInArgAndReturn"),
+			Entry(nil, "VariadicArgument"),
+			Entry(nil, "EmbeddedInterface"),
+			Entry(nil, "CustomInterfaceInArgAndReturn"),
+			Entry(nil, "EmptyInterface"),
+			Entry(nil, "InnerPackageReference"),
+			Entry(nil, "EmbedStandardInterface"),
+			Entry(nil, "EmbedCustomInterface"),
+			Entry(nil, "ComposeMultipleInterface"),
+			Entry(nil, "ExtendsAnotherInterface"),
+		)
+	})
+	Describe("duplicate packages", func() {
+		const (
+			pkgName        = "packagecollision"
+			testDirRelpath = testDataRelPath + "/" + pkgName
+
+			outDirRelPath       = testDirRelpath + "/" + generatedDirName
+			generatedDirRelPath = outDirRelPath + "/" + pkgName + "mw"
+
+			goldfilesDirRelPath = testDirRelpath + "/" + "goldfiles"
+		)
+		BeforeEach(func() {
+			// Comment out this line to keep all generated files after execution
+			// (e.g. if you want to use them as gold-files *after careful review*
+			Expect(cleanupGeneratedFiles(generatedDirRelPath)).To(Succeed())
+		})
+		It("aliases the duplicated package imports", func() {
+			interfaceName := "Compare"
+			a := app.New(testDirRelpath, interfaceName,
+				app.WithVersion("test"),
+				app.WithOutputDirectory(outDirRelPath),
+				app.WithLogger(log.New(GinkgoWriter, "", log.LstdFlags)),
+			)
+
+			err := a.Run()
+			Expect(err).ToNot(HaveOccurred())
+
+			fileName := interfaceName + ".go"
+			genFilePath := filepath.Join(generatedDirRelPath, snakeCase(fileName))
+			goldFilePath := filepath.Join(goldfilesDirRelPath, snakeCase(fileName))
+
+			compareFileContent(genFilePath, goldFilePath)
+		})
+	})
+})
+
+func compareFileContent(genFilePath, goldFilePath string) {
+	genFileContent, err := os.ReadFile(genFilePath)
+	Expect(err).ToNot(HaveOccurred())
+	goldFileContent, err := os.ReadFile(goldFilePath)
+	Expect(err).ToNot(HaveOccurred())
+
+	Expect(string(genFileContent)).To(Equal(string(goldFileContent)))
+}
+
+func cleanupGeneratedFiles(dir string) error {
+	files, err := ioutil.ReadDir(dir)
```

Comment:
`ioutil` is deprecated, use `os` or `io`, one of which should have the same method available

Reply:
Changed to `io`